clients/pkg/promtail/targets/syslog/syslogtarget.go

```diff
@@ -271,7 +271,14 @@ func (t *SyslogTarget) handleMessage(connLabels labels.Labels, msg syslog.Messag
 	} else {
 		timestamp = time.Now()
 	}
-	t.messages <- message{filtered, *rfc5424Msg.Message, timestamp}
+
+	m := *rfc5424Msg.Message
+	if t.config.UseRFC5424Message {
+		if fullMsg, err := rfc5424Msg.String(); err == nil {
```

Comment:
Maybe we should log a `debug` message here stating that the full message could not be retrieved. It'd be quite confusing if we enabled this setting and it had no effect because an error was being silently discarded.

Reply:
Good point!